### DISCUSSION TITLE
Not receiving more than two messages

### DISCUSSION THREAD

**Discussion by Greg-Lane** [monstrenyatko/ArduinoMqtt#29] [Q&A]
Hi 

I`m having trouble receiving more than 2 messages, the debug returns with "unexpected message", I think its because the receive buffers are full but I cant seem to clear them to receive further messages. any ideas on how? 
If anyone can point me to where in the libs where they get cleared I can probably sort it out myself.

Thanks Greg

**Comment by monstrenyatko** [comment]
Hi @Greg-Lane, the receive buffer is reused for each incoming message, you do not need to clean it, but make sure it is big enough to store a single message.
When the message is received the call-back is executed, you must process it or copy exactly at that moment as the memory/buffer is reused by another message.
